<a href="https://colab.research.google.com/github/prajjaldhar/DataScience/blob/main/Null_hypothesis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from scipy.stats import pearsonr

## Jo Zada padhta hai uske marks bhi zada ata hai.

In [ ]:
hours = [10,8,4,5,2,8,10,12,9]
marks = [45,52,68,75,88,78,56,48,66]
student={
    "hours":hours,
    "marks":marks
}
marks = pd.DataFrame(student)
marks

,hours,marks
0,10,45
1,8,52
2,4,68
3,5,75
4,2,88
5,8,78
6,10,56
7,12,48
8,9,66


In [ ]:
marks.describe()

,hours,marks
count,9.000000,9.000000
mean,7.555556,64.000000
std,3.244654,14.739403
min,2.000000,45.000000
25%,5.000000,52.000000
50%,8.000000,66.000000
75%,10.000000,75.000000
max,12.000000,88.000000


In [ ]:
marks.mean()

,0
hours,7.555556
marks,64.000000


## H0 : Hours and marks ke bich relationship exist karti nahi hai
## H1 : karti hai

In [ ]:
r,p_val=(pearsonr(marks["hours"],marks["marks"]))
print("Correlation: ",r)
print("p-value",p_val)

Correlation:  -0.8076445099809317
p-value 0.008467664648996854


In [ ]:
alpha = 0.05
if p_val<alpha:
  print("Reject H0")
else :
  print("Fail to reject H0")

Reject H0


Matlab is Sample mai hours and marks ke bich statistically linear relationship evidence show kar rahi hai
- Hours increase
- Marks decrease kar rahi hai
- corr = -0.808